lib/src/models/curl.dart

```diff
@@ -128,40 +150,51 @@ class Curl extends Equatable {
       }
     }
 
-    String? url = clean(result['url']);
-    final String? data = result['data'];
-    final String? cookie = result['cookie'];
-    final String? user = result['user'];
-    final String? referer = result['referer'];
-    final String? userAgent = result['user-agent'];
-    final bool form = result['form'] ?? false;
-    final bool head = result['head'] ?? false;
-    final bool insecure = result['insecure'] ?? false;
-    final bool location = result['location'] ?? false;
+    // Parse form data
+    List<List<String>>? formData;
+    if (result['form'] is List<String> && (result['form'] as List<String>).isNotEmpty) {
+      formData = [];
+      for (final formEntry in result['form']) {
+        final pairs = formEntry.split('=');
+        if (pairs.length != 2) {
+          throw Exception('Form data is not in key=value format');
+        }
+
+        // File type form data
+        if (pairs[1].startsWith('@')) {
+          pairs[1] = pairs[1].substring(1);
+        }
 
-    // Extract the request URL
-    url ??= result.rest.isNotEmpty ? clean(result.rest.first) : null;
+        formData.add([pairs[0], pairs[1]]);
+      }
+      headers ??= <String, String>{};
+      headers['Content-Type'] = 'multipart/form-data';
+    }
+
+    // Handle URL and query parameters
+    String? url = clean(result['url']) ?? (result.rest.isNotEmpty ? clean(result.rest.first) : null);
     if (url == null) {
-      throw Exception('url is null');
+      throw Exception('URL is null');
     }
     final uri = Uri.parse(url);
 
     return Curl(
-      method: head ? "HEAD" : (method ?? 'GET'),
+      method: result['head'] == true ? "HEAD" : (method ?? 'GET'),
       uri: uri,
       headers: headers,
-      data: data,
-      cookie: cookie,
-      user: user,
-      referer: referer,
-      userAgent: userAgent,
-      form: form,
-      insecure: insecure,
-      location: location,
+      data: result['data'],
+      cookie: result['cookie'],
+      user: result['user'],
+      referer: result['referer'],
+      userAgent: result['user-agent'],
+      form: formData != null && formData.isNotEmpty,
+      formData: formData,
+      insecure: result['insecure'] ?? false,
+      location: result['location'] ?? false,
```

Comment:
Accessing dict values during orbject creation is not preferred. Revert to using how it was done earlier.